libs/hdf-converters/src/converters-from-hdf/xccdf/reverse-xccdf-mapper.ts

```diff
@@ -0,0 +1,315 @@
+    ); // Splitting by underscore and checking for 'rule' is hardcoded into STIG Viewer
+    // Get control info
+    const group = {
+      id: control.data.id,
+      title: _.get(control.hdf.wraps.tags, 'gtitle', control.data.title),
+      control: {
+        id: controlId,
+        title: control.data.title || 'N/A',
+        identifiers: getReferences(control.data),
+        description: generateRuleDescription(control.data),
+        severity: extractTagOrDesc(control.hdf.wraps.descriptions, 'severity')
+          ? hdfImpactToXCCDFSeverity(control.data.impact)
+          : hdfSeverityToXCCDFSeverity(control.hdf.wraps.tags.severity),
+        checkContent:
+          extractTagOrDesc(control.hdf.wraps.descriptions, 'check') ||
+          extractTagOrDesc(control.hdf.wraps.tags, 'check') ||
+          'N/A',
+        checkId:
+          extractTagOrDesc(control.hdf.wraps.descriptions, 'checkid') ||
+          extractTagOrDesc(control.hdf.wraps.descriptions, 'check_id') ||
+          'N/A',
+        fixContent:
+          extractTagOrDesc(control.hdf.wraps.descriptions, 'fix') ||
+          extractTagOrDesc(control.hdf.wraps.tags, 'fix') ||
+          'N/A',
+        fixId:
+          extractTagOrDesc(control.hdf.wraps.tags, 'fixid') ||
+          extractTagOrDesc(control.hdf.wraps.tags, 'fix_id') ||
+          'N/A'
+      }
+    };
+    xccdfProfile?.groups.push(group);
+    // Get results info
+    const segments = getAllSegments(control);
+    if (segments.length >= 1) {
+      xccdfProfile?.results.push({
+        controlId: control.data.id,
+        severity: extractTagOrDesc(control.hdf.wraps.descriptions, 'severity')
+          ? hdfSeverityToXCCDFSeverity(control.hdf.wraps.tags.severity)
+          : hdfImpactToXCCDFSeverity(control.data.impact),
+        result: getStatus(segments),
+        identifiers: getReferences(control.data)
+      });
+    }
+  });
+  return xccdfProfile;
+}
+
+export function handleProfileJSON(hdf: ContextualizedProfile): XCCDFProfile {
+  const xccdfProfile: XCCDFProfile = {
+    title: hdf.data.title || 'N/A',
+    maintainer: hdf.data.maintainer || 'N/A',
+    groups: [],
+    date: new Date().toISOString().split('T')[0],
+    hasResults: false,
+    resultId: uuid(),
+    source: hdf.data.copyright_email || 'N/A',
+    results: []
+  };
+  hdf.contains.forEach((control) => {
+    const controlId = _.get(
+      control.data.tags,
+      'rid',
+      `${control.data.id}_rule`
+    ); // Splitting by underscore and checking for 'rule' is hardcoded into STIG Viewer
+    xccdfProfile?.groups.push({
+      id: control.data.id,
+      title: _.get(control.hdf.wraps.tags, 'gtitle', control.data.title),
+      control: {
+        id: controlId,
+        title: control.data.title || 'N/A',
+        identifiers: getReferences(control.data),
+        description: generateRuleDescription(control.data),
+        severity: hdfSeverityToXCCDFSeverity(control.hdf.wraps.tags.severity),
```

Comment:
why no check for impact here?  still seems to be a valid attribute in the profile type: https://github.com/mitre/inspecjs/blob/75348663388f79fa372b0883c016e4de41819c49/src/generated_parsers/v_1_0/profile-json.ts#L39

Reply:
but yeah please also see above comments on severity